yourbench/pipeline/answer_generation.py

```diff
@@ -0,0 +1,276 @@
+    if scenario == "with_correct_chunk":
+        # Load chunked dataset to retrieve correct chunk text
+        logger.info("Loading 'chunked' dataset to retrieve chunk text for with_correct_chunk scenario.")
+        try:
+            chunked_ds = custom_load_dataset(config, subset="chunked")
+            doc_meta_map = _build_doc_meta_map(chunked_ds)
+        except Exception as e:
+            logger.error(f"Failed loading chunked dataset for scenario={scenario}: {e}")
+            return calls, row_map
+
+    for i, row in enumerate(question_ds):
+        qtext = row.get("question", "")
+        if not qtext or not isinstance(qtext, str):
+            continue
+
+        if scenario == "zero_shot":
+            user_prompt = ZEROSHOT_QA_USER_PROMPT.format(question=qtext)
+            user_msg = {"role": "user", "content": user_prompt}
+            calls.append(InferenceCall(messages=[user_msg], tags=["zero_shot"]))
+            row_map.append(i)
+
+        elif scenario == "with_correct_chunk":
+            doc_id = row.get("document_id", "")
+            if not doc_id:
+                logger.debug(f"Row {i} missing document_id, skipping with_correct_chunk.")
+                continue
+
+            # Single-shot uses 'chunk_id', multi-hop uses 'source_chunk_ids'
+            if source_subset.startswith("single_shot"):
+                chunk_id = row.get("chunk_id", "")
+                chunk_text = doc_meta_map.get(doc_id, {}).get("chunks_map", {}).get(chunk_id, "")
+                doc_summary = doc_meta_map.get(doc_id, {}).get("document_summary", "")
+                user_prompt = GOLD_QA_USER_PROMPT.format(question=qtext, summary=doc_summary, document=chunk_text)
+                user_msg = {"role": "user", "content": user_prompt}
+                calls.append(InferenceCall(messages=[user_msg], tags=["with_correct_chunk"]))
+                row_map.append(i)
+
+            elif source_subset.startswith("multi_hop"):
+                chunk_ids = row.get("source_chunk_ids", [])
+                if not isinstance(chunk_ids, list):
+                    chunk_ids = []
+                doc_summary = doc_meta_map.get(doc_id, {}).get("document_summary", "")
+                combined_texts = []
+                for cid in chunk_ids:
+                    text_ = doc_meta_map.get(doc_id, {}).get("chunks_map", {}).get(cid, "")
+                    if text_:
+                        combined_texts.append(text_)
+                chunk_text_joined = "\n\n".join(combined_texts)
+
+                user_prompt = GOLD_QA_USER_PROMPT.format(
+                    question=qtext, summary=doc_summary, document=chunk_text_joined
+                )
+                user_msg = {"role": "user", "content": user_prompt}
+                calls.append(InferenceCall(messages=[user_msg], tags=["with_correct_chunk"]))
+                row_map.append(i)
+
+        else:
+            logger.warning(f"Unrecognized scenario '{scenario}' (row={i}), skipping.")
```

Comment:
Should fail way earlier as it will affect all rows

Reply:
(btw, no longer an issue if you invert the if and for clauses ^^)

Reply:
now raise a `ValueError` if the scenario is unrecognized (not in ("zero_shot","with_correct_chunk")) before processing any rows, instead of issuing repeated warnings per-row